### PR TITLE
Pull Addon

### PR DESCRIPTION
**Pull request by sean-e-dietrich**
Moving from Core into it's own addon

### CODE REVIEW

File: pull/README.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
+# Remote Provider Pull
+
+The `pull` addon was recently a core command but has been pulled out due as it is sometimes 
+quicker to get updates to just the addon as opposed to the whole platform.
```
Comment (shelane) [inline pull/README.md]: The `pull` addon was recently a core command but has been moved to an addon command.

File: pull/README.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+fin addon install --global pull
+```
+
+## Requirements
+
+### Acquia
+
+To interact with Acquia through Drush, the `SECRET_ACQUIA_CLI_SECRET` and
+`SECRET_ACQUIA_CLI_KEY` are required. Additionally, access to the Acquia Cloud API
+is required. If access is not allowed, fallback to using the [Drush](https://docs.docksal.io/fin/fin-pull/#drupal-drush) method.
```
Comment (shelane) [inline pull/README.md:20]: "To interact with Acquia through the Acquia CLI, ..."

Drush to Acquia only requires an SSH key. It's the Acquia CLI that requires the additional secret and key.

File: pull/README.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+More information on working with [Docksal with Acquia Drush](https://docs.docksal.io/tools/acquia-drush).
```
Comment (shelane) [inline pull/README.md]: More information on working with [Acquia CLI in Docksal](https://docs.docksal.io/tools/acquia-cli/).

File: pull/README.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+### Pantheon
+
+Interacting with Pantheon requires that the `SECRET_TERMINUS_TOKEN`
+is set. More information on [generating tokens](https://pantheon.io/docs/machine-tokens/)
+can be found in Pantheon's documentation.
+
+More information on working with [Docksal with Pantheon](https://docs.docksal.io/tools/terminus).
```
Comment (shelane) [inline pull/README.md]: [Pantheon in Docksal]

File: pull/README.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+### Platform.sh
+
+Interacting with Platform.sh requires that the `SECRET_PLATFORMSH_CLI_TOKEN`
+is set. For more information on [creating CLI tokens](https://docs.platform.sh/gettingstarted/cli/api-tokens.html)
+for use with the Platform.sh CLI tool, consult with Platform's documentation.
+
+More information on working with [Docksal with Platform.sh](https://docs.docksal.io/tools/platformsh).
```
Comment (shelane) [inline pull/README.md]: [Platform.sh in Docksal]

File: pull/README.md [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+Example of what `wp-cli.yml` would look like:
+
+```yaml
+@remote:
+  ssh: ssh-user@1.1.1.1/var/www/wordpress/docroot
+  url: http://example.com
+```
+
+More information on working with [Docksal and WP-CLI](https://docs.docksal.io/tools/wp-cli).
```
Comment (shelane) [inline pull/README.md]: [WP-CLI in Docksal]

**Review by shelane — CHANGES_REQUESTED** [review]
Some updated links and rewording of linking to the tools.

**Review by shelane — APPROVED** [review]

**Review by lmakarov — APPROVED** [review]

**Pull request merged**